### DISCUSSION TITLE
What happened to the expanded context window for Copilot?

### DISCUSSION THREAD

**Discussion by leahyra** [community/community#68008] [Copilot Conversations] [answered]
The [GitHub Copilot - August 28th Update](https://github.blog/changelog/2023-08-28-github-copilot-august-28th-update/) post on the GitHub blog's changelog stated that the Copilot context window had been increased to 8k. The Copilot Chat addon reflected this change in the VS Code output panel when interacting with the chat by printing `maxModelTokenWindow 16384`; however, since the beta was recently opened I've noticed that the addon has gone back to using the `copilot-chat` model which has a `maxModelTokenWindow` of 4096.

Why have you reneged on the increased 8k context window change?
> ### Expanding Copilot’s context window to 8k
> We’ve officially rolled out the 8k context window for all code completion requests! 🥳 With this change, Copilot has greater flexibility to include additional information as part of requests and ultimately improve the suggestions you receive!

**Comment by Akash1134** [comment]
Hi @leahyra ,

I have asked my team internally to share more details around your query.

Thanks for posting in the Community on Discussions.

**Comment by Akash1134** [comment, marked as answer]
Hi @leahyra ,

The Copilot Chat currently uses 4k tokens and the [GitHub Copilot - August 28th Update](https://github.blog/changelog/2023-08-28-github-copilot-august-28th-update/) is about the Copilot extension aimed at enhancing the code completion requests rate.

**Comment by richlysakowski** [comment]
I am now seeing a token length limit of 2048 when I generate Python.  This is MUCH worse than using Bard or ChatGPT for code generation.  What configuration can I change to get this up to the targeted 

I asked Copilot :  "what is the token length limit here when you generate Python?"

It responded:     "The token length limit for a single response from me is approximately 2048 tokens. A token can be as short as one character or as long as one word. For example, "a" is one token, and "apple" is also one token. This limit includes both the input from the user and the output from the assistant. If a conversation turn seems to be cut off, it may be because it has hit this length limit. When generating Python code, the same token limit applies."

I asked ChatGPT 4:  "What is the token limit for Python code generation for ChatGPT 4?"

It responded:     "The token limit for Python code generation in ChatGPT-4, including the code, comments, and any outputs, typically aligns with the overall token limit of a ChatGPT-4 response. This limit is approximately 4096 tokens in total. However, it's important to note that the specific limit can vary slightly based on the model configuration and the context of the conversation.

For Python code generation, the tokens are used for both the code itself and any textual output it generates. This means that longer code snippets or those producing extensive outputs might reach the token limit more quickly."

This is a very old post, so I will ask this on StackOverflow too.

**Reply by richlysakowski** [reply]
From the article above:  [token limit is 2048 tokens now](https://github.blog/changelog/2023-08-28-github-copilot-august-28th-update/)

"We’ve officially rolled out the 8k context window for all code completion requests! 🥳 With this change, Copilot has greater flexibility to include additional information as part of requests and ultimately improve the suggestions you receive!"

**Reply by zioalex** [reply]
Is this the same size used by the Copilot Chat functionality?

**Reply by Weblure** [reply]
First, Github Copilot and Github Copilot Chat are not the same thing. They are not running the same models, and they do not share the same context size.
Second, never ask an AI questions about itself. An LLM does not know what model its running, much less its context size, unless the prompt specifically provides this info (it doesn't). GPT4's minimum context window is 8k tokens, and can go up to 128k with some models.

**Comment by shadowyman** [comment]
Basically anyone looking for answers, this is worse help than chatgpt in terms of context window. The most absurd thing is they will let enterprise users train it on their data but no user can train it on their data. so never getting sleek answers or help. get ready to repeat yourself only to miss a critical line from code.

**Comment by sascharo** [comment]
Is the context window size still limited to only 2k?
